spyder/plugins/plots/plugin.py

```diff
@@ -4,154 +4,122 @@
-        layout.addWidget(self.stack)
-
-    def get_settings(self):
-        """Retrieve all Plots configuration settings."""
-        return {name: self.get_option(name) for name in
-                ['mute_inline_plotting', 'show_plot_outline',
-                 'auto_fit_plotting']}
-
-    # ---- Stack accesors
-    def set_current_widget(self, fig_browser):
-        """
-        Set the currently visible fig_browser in the stack widget, refresh the
-        actions of the cog menu button and move it to the layout of the new
-        fig_browser.
-        """
-        self.stack.setCurrentWidget(fig_browser)
-        # We update the actions of the options button (cog menu) and
-        # we move it to the layout of the current widget.
-        self._refresh_actions()
-        fig_browser.setup_options_button()
+    # --- SpyderDockablePlugin API
+    # ------------------------------------------------------------------------
+    def get_name(self):
+        return _('Plots')
 
+    def get_description(self):
+        return _('Display, explore and save console generated plots.')
+
+    def get_icon(self):
+        return self.create_icon('hist')
+
+    def register(self):
+        # Plugins
+        ipyconsole = self.get_plugin(Plugins.IPythonConsole)
+
+        # Signals
+        ipyconsole.sig_shellwidget_changed.connect(self.set_shellwidget)
+        ipyconsole.sig_shellwidget_process_started.connect(
+            self.add_shellwidget)
+        ipyconsole.sig_shellwidget_process_finished.connect(
+            self.remove_shellwidget)
+
+        # If a figure is loaded raise the dockwidget but do not give focus
+        self.get_widget().sig_figure_loaded.connect(
+            lambda: self.switch_to_plugin(force_focus=False))
+
+    def unregister(self):
+        # Plugins
+        ipyconsole = self.get_plugin(Plugins.IPythonConsole)
+
+        # Signals
+        ipyconsole.sig_shellwidget_id_changed.disconnect(
+            self.set_shellwidget_from_id)
+        ipyconsole.sig_shellwidget_process_started.disconnect(
+            self.add_shellwidget)
+        ipyconsole.sig_shellwidget_id_process_finished.disconnect(
+            self.remove_shellwidget_from_id)
+
+    # --- Public API
+    # ------------------------------------------------------------------------
     def current_widget(self):
-        return self.stack.currentWidget()
-
-    def count(self):
-        return self.stack.count()
-
-    def remove_widget(self, fig_browser):
-        self.stack.removeWidget(fig_browser)
+        """
+        Return the current shellwidget.
 
-    def add_widget(self, fig_browser):
-        self.stack.addWidget(fig_browser)
+        Returns
+        -------
+        spyder.plugins.plots.widgets.figurebrowser.FigureBrowser
+            The current widget.
 
-    # ---- Public API
-    def add_shellwidget(self, shellwidget):
+        shellwidget: spyder.plugins.ipyconsole.widgets.shell.ShellWidget
+            The shell widget.
```

Comment:
These two lines are unnecessary now. Please remove them.

Reply:
@goanpeca, this comment is still unaddressed. Please fix it so @dalthviz can give this one a last manual testing and merge it.

Reply:
Fixed